### DISCUSSION TITLE
Is Username & Password Publically Visible?

### DISCUSSION THREAD

**Discussion by Oliver-Turp** [PHPMailer/PHPMailer#2438] [Q&A]
Hi Guys,

Excuse the noob questions, I'm teaching myself this as I go.

Was running through a few tutorials and among other things to set everything up you need to enter login credentials (username and password). Was curious if these lines of code are publically visible when hosted?? If so is there a way to hide them/call them from another file that isn't publically hosted??

Sorry if this has been answered on the page already I didn't see anything ✌️

**Comment by Synchro** [comment]
No, your source code is not typically visible when hosted. However, there are numerous mechanisms to help you not have credentials stored directly in your code, such as `.env` files, environment variables, and numerous dedicated systems for managing secrets (which is the name for what you're asking about). Even then, it's common for all the PHP code you run on a server to be owned by the same user, and so that user can read all the files. This is the kind of thing that is handled by privilege separation, creation of multiple users, etc – but notice that these are concerned with the environment your scripts run within rather than the code you're running, as that's the right place to protect such things.
